doc/developers_guide/developers_guide.md

```diff
@@ -0,0 +1,44 @@
+# Developers Guide
+
+This guide contains information for developers.
+
+## Working With the Extension
```

Comment:
Not sure whether the term "Extension" is self-explaining here. 
Should we add an introduction passage and/or a link?
How about a term like "Managed Extension" or "Plugin" or "Integration with Exasol and Exasol SAAS"?

Reply:
Potentially also the descriptions for "using a local, non-published..." could turn out to be generic and if so, may be moved to the extension manager developers guide or another central place.

Reply:
Added a link to the extension manager